sdk/storage/storage-file-share/src/Clients.ts

```diff
@@ -4110,16 +4183,22 @@ export class ShareFileClient extends StorageClient {
     // FileAttributes, filePermission, createTime, lastWriteTime will all be preserved.
     options = validateAndSetDefaultsForFileAndDirectorySetPropertiesCommonOptions(options);
     return tracingClient.withSpan("ShareFileClient-resize", options, async (updatedOptions) => {
-      return assertResponse<FileSetHttpHeadersHeaders, FileSetHttpHeadersHeaders>(
-        await this.context.setHttpHeaders(fileAttributesToString(updatedOptions.fileAttributes!), {
-          ...updatedOptions,
-          fileContentLength: length,
-          fileChangeOn: fileChangeTimeToString(options.changeTime),
-          fileCreatedOn: fileCreationTimeToString(options.creationTime),
-          fileLastWriteOn: fileLastWriteTimeToString(options.lastWriteTime),
-          ...this.shareClientConfig,
-        }),
-      );
+      const rawResponse = await this.context.setHttpHeaders({
+        ...updatedOptions,
+        fileContentLength: length,
+        fileChangeOn: fileChangeTimeToString(options.changeTime),
+        fileCreatedOn: fileCreationTimeToString(options.creationTime),
+        fileLastWriteOn: fileLastWriteTimeToString(options.lastWriteTime),
+        fileAttributes: fileAttributesToString(updatedOptions.fileAttributes!),
+        owner: updatedOptions.posixProperties?.owner,
+        group: updatedOptions.posixProperties?.group,
```

Comment:
Should resize support Filemode?
https://github.com/Azure/azure-rest-api-specs/blob/1d424c359209fda4d6ec397e43267da82e87a070/specification/storage/data-plane/Microsoft.FileStorage/stable/2025-05-05/file.json#L4507